dali/operators/image/distortion/jpeg_compression_distortion_op_cpu.cc

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include <opencv2/opencv.hpp>
+#include "dali/operators/image/distortion/jpeg_compression_distortion_op.h"
+
+namespace dali {
+
+DALI_SCHEMA(JpegCompressionDistortion)
+    .DocStr(R"code(Produces JPEG-like distortion in RGB images.
+
+The level of degradation of the image can be controlled with the ``quality`` argument,
+)code")
+    .NumInput(1)
+    .InputLayout(0, "HWC")
```

Comment:
How hard would it be to reorganize it to support leading non-channel dimensions (frames, 3D slices) so we could simulate MJPEG, for example?

Reply:
It's certainly doable. We just need to create a flattened view of the batch, where each frame or slice is a sample.

Reply:
I created a task to be considered later: DALI-1951